charts/kong/templates/_helpers.tpl

```diff
@@ -59,29 +59,37 @@ Generic tool for creating KONG_PROXY_LISTEN, KONG_ADMIN_LISTEN, etc.
 {{- define "kong.listen" -}}
   {{- $unifiedListen := list -}}
 
-  {{- if .http.enabled -}}
-    {{- $listenConfig := dict -}}
-    {{- $listenConfig := merge $listenConfig .http -}}
-    {{- $_ := set $listenConfig "address" (default "0.0.0.0" .address) -}}
-    {{- $httpListen := (include "kong.singleListen" $listenConfig) -}}
-    {{- $unifiedListen = append $unifiedListen $httpListen -}}
+  {{/* Some services do not support these blocks at all, so these checks are a
+       two-stage "is it safe to evaluate this?" and then "should we evaluate
+       this?"
+  */}}
+  {{- if .http -}}
+    {{- if .http.enabled -}}
```

Comment:
nit: make this simpler with `if and .http .http.enabled`

Reply:
gotemplate attempts to check both always if they're within an `and` rather than failing fast if the first condition isn't met, so it null errors if this isn't nested.